### ISSUE TITLE
compiling fiberassign on an OS X laptop?

### ISSUE THREAD

**Issue by moustakas** [desihub/fiberassign#117]
Has anybody been able to compile `fiberassign` on a Macbook Pro (running 10.13.4 with updated Xcode)? 

```bash
(desi) ioannis@atlas:~/repos/desihub/fiberassign% make
make: Entering directory `/Users/ioannis/repos/desihub/fiberassign'
make[1]: Entering directory `/Users/ioannis/repos/desihub/fiberassign/src'
clang++ -O3  -I.  -o fiberassign.o -c fiberassign.cpp
In file included from fiberassign.cpp:16:
./structs.h:26:14: warning: in-class initialization of non-static data member is a C++11 extension [-Wc++11-extensions]
    int stuck=0; // 0=OK
             ^
./structs.h:27:15: warning: in-class initialization of non-static data member is a C++11 extension [-Wc++11-extensions]
    int broken=0; // 0=OK
              ^
fiberassign.cpp:51:33: warning: format specifies type 'int' but the argument has type 'std::__1::vector<target,
      std::__1::allocator<target> >::size_type' (aka 'unsigned long') [-Wformat]
    printf(" Target size %d \n",M.size());
                         ~~     ^~~~~~~~
                         %lu
fiberassign.cpp:69:40: warning: format specifies type 'int' but the argument has type 'std::__1::vector<target,
      std::__1::allocator<target> >::size_type' (aka 'unsigned long') [-Wformat]
    printf(" Standard Star size %d \n",M.size());
                                ~~     ^~~~~~~~
                                %lu
fiberassign.cpp:71:36: warning: format specifies type 'int' but the argument has type 'std::__1::vector<target,
      std::__1::allocator<target> >::size_type' (aka 'unsigned long') [-Wformat]
    printf(" Sky Fiber size %d \n",M.size());
                            ~~     ^~~~~~~~
                            %lu
5 warnings generated.
clang++ -O3  -I.  -o collision.o -c collision.cpp
collision.cpp:158:1: warning: control may reach end of non-void function [-Wreturn-type]
}
^
1 warning generated.
clang++ -O3  -I.  -o feat.o -c feat.cpp
clang++ -O3  -I.  -o global.o -c global.cpp
In file included from global.cpp:23:
./structs.h:26:14: warning: in-class initialization of non-static data member is a C++11 extension [-Wc++11-extensions]
    int stuck=0; // 0=OK
             ^
./structs.h:27:15: warning: in-class initialization of non-static data member is a C++11 extension [-Wc++11-extensions]
    int broken=0; // 0=OK
              ^
global.cpp:66:27: error: non-aggregate type 'std::vector<double>' cannot be initialized with an initializer list
              std::vector<double>posit={pp[k].fp_x,pp[k].fp_y};
                                 ^     ~~~~~~~~~~~~~~~~~~~~~~~
global.cpp:313:40: error: a space is required between consecutive right angle brackets (use '> >')
       std::vector<std::pair<double,int>> pairs;
                                       ^~
                                       > >
2 warnings and 2 errors generated.
make[1]: *** [global.o] Error 1
make[1]: Leaving directory `/Users/ioannis/repos/desihub/fiberassign/src'
make: *** [all] Error 2
make: Leaving directory `/Users/ioannis/repos/desihub/fiberassign'
```

**Comment by forero** [comment]
I've managed to compile it on a macbook pro with macos sierra 10.12.6.

This setup works for me in `platforms/osx`:

```bash

# The serial C++ compiler
export CXX := clang++

# Normal compile flags
export CXXFLAGS := -O3 -std=gnu++11

# OpenMP flags (used for both compiling and linking)
export OMPFLAGS := 

# CFITSIO include and linking commands
export CFITSIO_CPPFLAGS := -I/opt/local/include
export CFITSIO := -L/opt/local/lib -lcfitsio -lm
```

**Comment by sbailey** [comment]
@moustakas note the instructions in the README are a bit more than the standard "make".  In your case I think you want "make PLATFORM=osx install".

**Comment by moustakas** [comment]
Thanks @sbailey I did see the README but without @forero's suggested modifications to the `platforms/osx` file the code still doesn't compile.  In the end using the following flags worked for me:
```
# The serial C++ compiler                                                                                                  
export CXX := clang++                                                                                                      
                                                                                                                           
# Normal compile flags                                                                                                     
export CXXFLAGS := -O3 -std=gnu++11                                                                                        
                                                                                                                           
# OpenMP flags (used for both compiling and linking)                                                                       
export OMPFLAGS :=                                                                                                         
                                                                                                                           
# CFITSIO include and linking commands                                                                                     
export CFITSIO_CPPFLAGS :=                                                                                                 
export CFITSIO := -lcfitsio -lm                                                                                            
```

I leave it up to the lead developers to decide whether these changes should be incorporated in a PR and/or whether to close this issue.

**Comment by tskisner** [comment]
Looks like this means that the code is requiring GNU extensions.  Short term solution is to edit OSX platform file.  Longer term we could remove things that require GNU extensions.

**Comment by tskisner** [comment]
This is fixed in PR #124.

**Comment by tskisner** [comment]
Closed by #124
